addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1287,6 +1287,10 @@ var SnippetEditor = Widget.extend({
                     self.dragState.backgroundGridEl = backgroundGridEl;
                     self.onDragMove = self._onDragMove.bind(self);
                     self.$body[0].addEventListener('mousemove', self.onDragMove, false);
+                    // Triggering first move right away without waiting for a
+                    // mousemove event.
+                    const cursorAt = self.draggableComponent.options.jQueryDraggableOptions.cursorAt;
+                    self._dragMove(ui.offset.left + cursorAt.left, ui.offset.top + cursorAt.top);
```

Comment:
In fact, sorry I didn't think of it because since you were using the `originalEvent` before, I thought the positions were not easy to find, but can't they simply be retrieved from `ev` (instead of `ui` which then needs to be adjusted) ? 

For the record, I wanted to see if directly writing `ui` as the argument (and remove `ev`) would still work (-> obviously, it doesn't :sweat_smile: ) so I printed them and then I noticed that there are the `pageX` and `pageY` stored in `ev` (and without the `cursorAt` offset) :open_mouth: 
![image](https://github.com/odoo/odoo/assets/102796037/c22b8466-4752-46ec-89bc-36fae7dbd83a)

![image](https://github.com/odoo/odoo/assets/102796037/a517fc47-8db2-4873-8ae4-dbc1b4a1adab)

Am I missing something that prevents us to use them directly ? :thinking: 

It would have the advantage to not need to retrieve `cursorAt` :slightly_smiling_face: And `onDragMove` could have directly been called with this same `ev`, without needing to extract it :thinking:

Reply:
> And `onDragMove` could have directly been called with this same `ev`, without needing to extract it

But IMO that would be a kind of "language abuse", the two events are of different natures - it works "by coïncidence". I'd rather keep it in a separated method to be safe - otherwise an override of `onDragMove` might be tempted to access other [possibly future] fields of the drag move events which might not be part of the drag start event.